Index: financial-system/kfs/branches/mod-jhkeller/KITT-1750/work/src/edu/arizona/kfs/module/ld/batch/service/impl/LaborEncumbranceAdjustmentServiceImpl.java
===================================================================
--- financial-system/kfs/branches/mod-jhkeller/KITT-1750/work/src/edu/arizona/kfs/module/ld/batch/service/impl/LaborEncumbranceAdjustmentServiceImpl.java	(revision 14486)
+++ financial-system/kfs/branches/mod-jhkeller/KITT-1750/work/src/edu/arizona/kfs/module/ld/batch/service/impl/LaborEncumbranceAdjustmentServiceImpl.java	(revision 14487)
@@ -216,7 +216,7 @@
                             if ( employeePerson == null ) {
                                 LOG.warn( "No Person found for employee ID: " + reversingOriginEntry.getEmplid() );
                             } else {
-                                newDescription = employeePerson.getNameUnmasked();
+                                newDescription = (employeePerson.getLastNameUnmasked() + ", " + employeePerson.getFirstNameUnmasked() + " " + employeePerson.getMiddleNameUnmasked()).trim();
                             }
                         } catch ( Exception ex ) {
                             LOG.warn( "Error Obtaining Person for employee ID: " + reversingOriginEntry.getEmplid(), ex );
